staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -328,6 +338,25 @@ func (g *Cloud) ensureInternalFirewall(svc *v1.Service, fwName, fwDesc string, s
 	if err != nil && !isNotFound(err) {
 		return err
 	}
+	if legacyFwName != "" {
```

Comment:
maybe delete the legacy Firewall after the new one gets created?

Reply:
fixed.